src/instructlab/cli/model/upload.py

```diff
@@ -0,0 +1,249 @@
+                    fg="red",
+                )
+                raise click.exceptions.Exit(1)
+
+        # upload either safetensors or gguf, if valid
+        if is_model_safetensors(self.local_model_path):
+            self.upload_safetensors()
+        elif is_model_gguf(self.local_model_path):
+            self.upload_gguf()
+        else:
+            click.secho(
+                f"Local model path {self.local_model_path} is a valid path, but is neither safetensors nor a GGUF - cannot upload to Hugging Face",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def upload_gguf(self) -> None:
+        try:
+            hf_api = HfApi()
+            resp = hf_api.upload_file(
+                path_or_fileobj=self.local_model_path,
+                path_in_repo=self.model,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+            logger.debug(f"Hugging Face response:\n{resp}")
+        except Exception as exc:
+            click.secho(
+                f"\nUploading GGUF model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            "\nUploading GGUF model succeeded!",
+            fg="green",
+        )
+
+    def upload_safetensors(self) -> None:
+        try:
+            hf_api = HfApi()
+            resp = hf_api.upload_folder(
+                folder_path=self.local_model_path,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+            logger.debug(f"Hugging Face response:\n{resp}")
+        except Exception as exc:
+            click.secho(
+                f"\nUploading safetensors model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            "\nUploading safetensors model succeeded!",
+            fg="green",
+        )
+
+
+@click.command()
+@click.option(
+    "--model",
+    "-m",
+    required=True,
+    help="Local model to upload.",
+)
+@click.option(
+    "--model-dir",
+    default=lambda: DEFAULTS.CHECKPOINTS_DIR,
+    show_default="The default system model location store, located in the data directory.",
+    help="The local directory to upload the model files from.",
+)
+@click.option(
+    "--dest-type",
+    type=click.Choice(["hf", "oci", "s3"]),
+    required=True,
+    help="The type of destination to upload to - can be 'hf', 'oci', or 's3'",
+)
+@click.option(
+    "--upload-dest",
+    type=str,
+    required=True,
+    help="Destination for the model to be uploaded to. Ex: for a Hugging Face repo, should be '<username>/<repo-name>'",
+)
+@click.option(
+    "--release",
```

Comment:
This was mostly to align with terminology Hugging Face uses as well as `ilab model download`

It will also be generalized to other upload targets (e.g. a tag for an OCI endpoint)

Reply:
Makes sense, carry on!